src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -0,0 +1,371 @@
+            const selectedElement = document.querySelector(this.currentTour.steps[this.currentTourStepIndex].selector!);
+            if (!selectedElement) {
+                this.errorHandler.handleError(
+                    // If error handler is configured this should not block the browser.
+                    new Error(
+                        `Error finding selector ${this.currentTour.steps[this.currentTourStepIndex].selector} on step ${this.currentTourStepIndex + 1} during guided tour: ${
+                            this.currentTour.settingsId
+                        }`,
+                    ),
+                );
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * @return {boolean} if the current step is the last tour step
+     */
+    public get isOnLastStep(): boolean {
+        if (!this.currentTour) {
+            return false;
+        }
+        return this.currentTour.steps.length - 1 === this.currentTourStepIndex;
+    }
+
+    /**
+     * @return {boolean} if the current step is the first tour step
+     */
+    public get isOnFirstStep(): boolean {
+        return this.currentTourStepIndex === 0;
+    }
+
+    /**
+     * @return {boolean} if the `show resize` message should be displayed
+     */
+    public get isOnResizeMessage(): boolean {
+        return this.onResizeMessage;
+    }
+
+    /**
+     * @return current tour step number
+     */
+    public get currentTourStepDisplay(): number {
+        return this.currentTourStepIndex + 1;
+    }
+
+    /**
+     *  @return total count of tour steps of the current tour
+     */
+    public get currentTourStepCount(): any {
+        return this.currentTour && this.currentTour.steps ? this.currentTour.steps.length : 0;
+    }
+
+    /**
+     *  Prevents the tour from advancing by clicking the backdrop
+     *  @return {boolean} `preventBackdropFromAdvancing` configuration if tour should advance when clicking on the backdrop
+     *  or false if this configuration is not set
+     */
+    public get preventBackdropFromAdvancing(): boolean {
+        if (this.currentTour) {
+            return this.currentTour && (this.currentTour.preventBackdropFromAdvancing ? this.currentTour.preventBackdropFromAdvancing : false);
+        }
+        return false;
+    }
+
+    /**
+     * Get the tour step with defined orientation
+     * @param index current tour step index
+     * @return prepared current tour step or null
+     */
+    private getPreparedTourStep(index: number): TourStep | null {
+        if (this.currentTour) {
+            return this.setTourOrientation(this.currentTour.steps[index]);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Set orientation of the passed on tour step
+     * @param {step} passed on tour step of a guided tour
+     * @return guided tour step with defined orientation
+     */
+    private setTourOrientation(step: TourStep): TourStep {
```

Comment:
This is a good example for a method that could be Unit tested:
Input -> Output.